lib/stemcell/launcher.rb

```diff
@@ -186,14 +186,25 @@ def launch(opts={})
       instances = do_launch(launch_options)
 
       # set tags on all instances launched
-      set_tags(instances, tags)
-      @log.info "sent ec2 api requests successfully"
-
-      # wait for aws to report instance stats
-      if opts.fetch('wait', true)
-        wait(instances)
-        print_run_info(instances)
-        @log.info "launched instances successfully"
+      begin
+        set_tags(instances, tags)
+        @log.info "sent ec2 api requests successfully"
+
+        # wait for aws to report instance stats
+        if opts.fetch('wait', true)
+          wait(instances)
+          print_run_info(instances)
+          @log.info "launched instances successfully"
+        end
+      rescue => e
+        @log.info "launch failed, killing all launched instances"
+        # ensure the propery cleanup
```

Comment:
Minor typo in comment. I think we can just do without the comment since the log message is pretty clear.